q2_phylogenomics/plugin_setup.py

```diff
@@ -0,0 +1,108 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from itertools import permutations, chain
+from qiime2.plugin import (
+    Choices,
+    Plugin,
+    Citations,
+    Range,
+    Int,
+    Str,
+)
+from q2_types.sample_data import SampleData
+from q2_types.per_sample_sequences import (
+    SequencesWithQuality,
+    PairedEndSequencesWithQuality,
+)
+
+import q2_phylogenomics
+import q2_phylogenomics._prinseq
+
+
+citations = Citations.load('citations.bib', package='q2_phylogenomics')
+
+plugin = Plugin(
+    name='phylogenomics',
+    version=q2_phylogenomics.__version__,
+    website='https://github.com/qiime2/q2-phylogenomics',
+    package='q2_phylogenomics',
+    description='A QIIME 2 plugin for phylogenomics analyses.',
+    short_description='A QIIME 2 plugin for phylogenomics analyses.',
+)
+
+prinseq_input = {'demultiplexed_sequences': 'The sequences to be trimmed.'}
+prinseq_output = {'trimmed_sequences': 'The resulting trimmed sequences.'}
+
+# valid derep modes include any combination of 1, 2, 3, 4, and 5
+derep_choices = list(chain.from_iterable(
+    [[''.join(i) for i in permutations(list('12345'), int(x))]
+     for x in list('12345')]))
+
+prinseq_parameters = {
+    'trim_qual_right': Int % Range(1, None),
+    'trim_qual_type': Str % Choices(['min', 'mean', 'max', 'sum']),
+    'trim_qual_window': Int % Range(1, None),
+    'min_qual_mean': Int % Range(1, None),
+    'min_len': Int % Range(1, None),
+    'lc_method': Str % Choices(['dust', 'entropy']),
+    'lc_threshold': Int % Range(0, 100),
+    'derep': Str % Choices(derep_choices)}
```

Comment:
This is probably cleaner as a `List[Str % Choices("1", "2, "3, "4", "5")]` although it can't handle redundancy, and `Set` would lose order... But that still seems a bit nicer than figuring out the Choices pattern and realizing it's just 1,2,3,4,5 in some order.

We could also do `List[Int % Range(1, 5, inclusive_end=True)]` to make it easier to type.

Reply:
good call on the List. Much cleaner. I had considered that earlier but thought it "broke" the prinseq usage too much... but after your review I see that's so much better than the ugliness I created.